.github/workflows/tc-asa-build-and-test.yaml

```diff
@@ -0,0 +1,51 @@
+name: TC ASA Build and Test
+
+on: [pull_request]
+
+jobs:
+  build-backend:
+    name: Build & Test TC ASA
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.9.10
+
+    - name: Install pipenv
+      run: |
+        python -m pip install --upgrade pipenv wheel
+    - id: cache-pipenv
+      uses: actions/cache@v1
+      with:
+        path: ~/.local/share/virtualenvs
+        key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
+
+    - name: Install dependencies
+      if: steps.cache-pipenv.outputs.cache-hit != 'true'
+      run: |
+        pipenv install --deploy --dev
+
+    - uses: pre-commit/action@v2.0.3
+      name: "Linters and formatters check"
+      with:
+        extra_args: --all-files
+
+    - name: Clone Algorand Sanbox
```

Comment:
Do we wanna consider using the images from `https://github.com/MakerXStudio/algorand-sandbox-dev` instead of the sandbox?